rcl/test/rcl/allocator_testing_utils.h

```diff
@@ -87,6 +87,80 @@ set_failing_allocator_is_failing(rcutils_allocator_t & failing_allocator, bool s
   ((__failing_allocator_state *)failing_allocator.state)->is_failing = state;
 }
 
+typedef struct time_bomb_allocator_state
+{
+  int count_until_failure;
+} time_bomb_allocator_state;
+
+void * time_bomb_malloc(size_t size, void * state)
+{
+  time_bomb_allocator_state * time_bomb_state = (time_bomb_allocator_state *)state;
+  if (time_bomb_state->count_until_failure >= 0 &&
+    time_bomb_state->count_until_failure-- == 0)
+  {
+    return nullptr;
+  }
+  return rcutils_get_default_allocator().allocate(size, rcutils_get_default_allocator().state);
+}
+
+void *
+time_bomb_realloc(void * pointer, size_t size, void * state)
+{
+  time_bomb_allocator_state * time_bomb_state = (time_bomb_allocator_state *)state;
+  if (time_bomb_state->count_until_failure >= 0 &&
+    time_bomb_state->count_until_failure-- == 0)
+  {
+    return nullptr;
+  }
+  return rcutils_get_default_allocator().reallocate(
+    pointer, size, rcutils_get_default_allocator().state);
+}
+
+void
+time_bomb_free(void * pointer, void * state)
+{
+  time_bomb_allocator_state * time_bomb_state = (time_bomb_allocator_state *)state;
+  if (time_bomb_state->count_until_failure >= 0 &&
+    time_bomb_state->count_until_failure-- == 0)
+  {
+    return;
+  }
+  rcutils_get_default_allocator().deallocate(pointer, rcutils_get_default_allocator().state);
+}
+
+void *
+time_bomb_calloc(size_t number_of_elements, size_t size_of_element, void * state)
+{
+  time_bomb_allocator_state * time_bomb_state = (time_bomb_allocator_state *)state;
+  if (time_bomb_state->count_until_failure >= 0 &&
+    time_bomb_state->count_until_failure-- == 0)
+  {
+    return nullptr;
+  }
+  return rcutils_get_default_allocator().zero_allocate(
+    number_of_elements, size_of_element, rcutils_get_default_allocator().state);
+}
+
+static inline rcutils_allocator_t
+get_time_bombed_allocator(void)
+{
+  time_bomb_allocator_state state;
+  state.count_until_failure = 1;
+  auto time_bombed_allocator = rcutils_get_default_allocator();
+  time_bombed_allocator.allocate = time_bomb_malloc;
+  time_bombed_allocator.deallocate = time_bomb_free;
+  time_bombed_allocator.reallocate = time_bomb_realloc;
+  time_bombed_allocator.zero_allocate = time_bomb_calloc;
+  time_bombed_allocator.state = &state;
```

Comment:
so, state goes out of scope here, and you're storing a pointer to it.
Then, you will have use the pointer when it's already invalid.

I see why you were using an static, but IMO that's a bit hacky.
I think that the best thing would be to allocate the state in the heap, and then deallocate it when we're done with the allocator

Reply:
I agree it's a bit hacky, but as it is a testing util I personally don't think the extra heap management is required

Reply:
> I agree it's a bit hacky, but as it is a testing util I personally don't think the extra heap management is required

Well, you can readd the `static` keyword in the state if you want.
But the current code is broken, because you're storing a pointer to an object that not longer exists.

Reply:
I added it again, the code CI was all red because I set up badly the job configuration